lib/msf/ui/console/command_dispatcher/modules.rb

```diff
@@ -971,6 +975,142 @@ def cmd_back(*args)
+            print_line '  -D                Delete the fav_modules file'
+          end
+
+          #
+          # Helper method for cmd_favorite that writes modules to the fav_modules_file
+          #
+          def cmd_favorite_add(modules, favs_file, contents)
+            modules = modules.uniq # prevent modules from being added more than once
+            modules.each do |name|
+              mod = framework.modules.create(name)
+              if (mod == nil)
+                print_error("Invalid module: #{name}")
+                next
+              end
+
+              if contents && contents.include?(mod.fullname)
+                print_warning("Module #{mod.fullname} has already been favorited and will not be added to #{favs_file}")
+                next
+              end
+
+              begin
+                File.open(favs_file, 'a+') { |file| file.puts(mod.fullname) }
+              rescue Errno::EACCES => e
+                print_error("Unable to write to #{favs_file}").
+                return
+              end
+              print_status("Added #{mod.fullname} to #{favs_file}")
+            end
+          end
+
+          #
+          # Helper method for cmd_favorite that deletes modules from the fav_modules_file, or deletes the file altogether
+          #
+          def cmd_favorite_del(modules, delete_all, favs_file, contents)
+            if delete_all
+              print_status("Removing #{favs_file} if it exists")
+              File.delete(favs_file) if File.exist?(favs_file)
+              return
+            end
+
+            modules = modules.uniq # prevent modules from being deleted more than once
+            contents = contents.split
+            modules.each do |name|
+              mod = framework.modules.create(name)
+              if (mod == nil)
+                print_error("Invalid module: #{name}")
+                next
+              end
+
+              unless contents.include?(mod.fullname)
+                print_warning("Module #{mod.fullname} cannot be deleted because it is not in #{favs_file}")
+                next
+              end
+
+              # remove module from contents
+              contents.delete(mod.fullname)
+              print_status("Removing #{mod.fullname} from #{favs_file}")
+            end
+
+            begin
+              # delete fav_modules file if removing the module(s) makes it empty
+              if contents.length == 0
+                File.delete(favs_file)
+                print_status("Removing #{favs_file} since it is empty now")
+                return
+              end
+              File.open(favs_file, 'w') { |file| file.puts(contents.join("\n")) }
+            rescue Errno::EACCES => e
+              print_error("Unable to write to #{favs_file}").
+              return
+            end
+          end
+
+          #
+          # Add modules to the fav_modules file
+          #
+          def cmd_favorite(*args)
```

Comment:
If the `Rex::Parser::Arguments` class is used for the command options, the `parse()` method can be used with `args` to simplify the processing of the arguments:

```ruby
@@favorite_opts.parse(args) do |opt, i, val|
  case opt
  when '-d'
    ...
  when '-D'
    ...
  when '-h'
    print @@favorite_opts.usage
    return
  end
end
```

Reply:
Done and done!